hera_cal/lstbin.py

```diff
@@ -1000,22 +1002,24 @@ def sigma_clip(array, flags=None, sigma=4.0, axis=0, min_N=4):
     # create empty clip_flags array
     clip_flags = np.zeros_like(array, np.bool)
 
+    # make copy of data array as to not replace original array
+    array_ = array.copy()
+
     # inherit flags if fed and apply flags to data
     if flags is not None:
         clip_flags += flags
-        array[flags] *= np.nan
+        array_[flags] *= np.nan
 
     # get robust location
-    location = np.nanmedian(array, axis=axis)
+    location = np.nanmedian(array_, axis=axis)
 
     # get MAD! * 1.482579
-    scale = np.nanmedian(np.abs(array - location), axis=axis) * 1.482579
+    scale = np.nanmedian(np.abs(array_ - location), axis=axis) * 1.482579
 
     # get clipped data
-    clip = np.abs(array - location) / scale > sigma
+    clip = np.abs(array_ - location) / scale > sigma
```

Comment:
Can't we just return clip instead of doing `clip_flags[clip] = True`?

Reply:
`if flags is not None` then `clip != clip_flags` so could leave as is?